crud/src/main/java/com/redhat/lightblue/eval/SortableItem.java

```diff
@@ -38,8 +38,10 @@ public SortableItem(JsonNode node,SortFieldInfo[] sortFields) {
         this.sortFields=sortFields;
         keyValues=new Object[sortFields.length];
         for(int i=0;i<sortFields.length;i++) {
-            JsonNode valueNode=JsonDoc.get(node,sortFields[i].getName());
-            keyValues[i]=sortFields[i].getField().getType().fromJson(valueNode);
+            if (sortFields[i] != null) {
```

Comment:
That should not be. This change will probably mask another bug by working
around it.
On Dec 8, 2015 2:27 PM, "Naveen Malik" <notifications@github.com> wrote:

> In crud/src/main/java/com/redhat/lightblue/eval/SortableItem.java
> <https://github.com/lightblue-platform/lightblue-core/pull/537#discussion_r47016077>
> :
>
> > @@ -38,8 +38,10 @@ public SortableItem(JsonNode node,SortFieldInfo[] sortFields) {
> >          this.sortFields=sortFields;
> >          keyValues=new Object[sortFields.length];
> >          for(int i=0;i<sortFields.length;i++) {
> > -            JsonNode valueNode=JsonDoc.get(node,sortFields[i].getName());
> > -            keyValues[i]=sortFields[i].getField().getType().fromJson(valueNode);
> > +            if (sortFields[i] != null) {
>
> Don't know, I just hit this with an internal client and fixed the two
> places it broke.
> … <#151837e15925f5ca_>
> On Tue, Dec 8, 2015 at 12:57 PM, Burak Serdar <notifications@github.com>
> wrote: In crud/src/main/java/com/redhat/lightblue/eval/SortableItem.java <#537
> (comment)
> <https://github.com/lightblue-platform/lightblue-core/pull/537#discussion_r46988558>>
> : > @@ -38,8 +38,10 @@ public SortableItem(JsonNode node,SortFieldInfo[]
> sortFields) { > this.sortFields=sortFields; > keyValues=new
> Object[sortFields.length]; > for(int i=0;i<sortFields.length;i++) { > -
> JsonNode valueNode=JsonDoc.get(node,sortFields[i].getName()); > -
> keyValues[i]=sortFields[i].getField().getType().fromJson(valueNode); > + if
> (sortFields[i] != null) { Why is there an array of SortFields with null in
> it? — Reply to this email directly or view it on GitHub <
> https://github.com/lightblue-platform/lightblue-core/pull/537/files#r46988558>
> .
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lightblue-platform/lightblue-core/pull/537/files#r47016077>
> .
>